File: Python/28.实现 strStr() 暴力解法.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for i in range(0, len(haystack) - len(needle) + 1):
            match: bool = True
            for j in range(0, len(needle)):
                if i + j == len(haystack):
                    return -1
                if haystack[i + j] != needle[j]:
                    match = False
                    break
            if match:
                return i
        return -1

File: TypeScript/28.实现 strStr() 暴力解法.ts
function strStr(haystack: string, needle: string): number {
    for (let i = 0; i + needle.length <= haystack.length; ++i) {
        let match: boolean = true;
        for (let j = 0; j < needle.length; ++j) {
            if (i + j == haystack.length) {
                return -1;
            }
            if (haystack[i + j] != needle[j]) {
                match = false;
                break;
            }
        }
        if (match) {
            return i;
        }
    }
    return -1;
};
